app/src/display/widgets/mods_status.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/display.h>
+#include <zmk/display/widgets/mods_status.h>
+#include <zmk/events/keycode_state_changed.h>
+#include <zmk/event_manager.h>
+#include <zmk/endpoints.h>
+#include <zmk/hid.h>
+
+static sys_slist_t widgets = SYS_SLIST_STATIC_INIT(&widgets);
+
+struct mods_status_state {
+    uint8_t mods;
+};
+
+struct mods_status_state mods_status_get_state(const zmk_event_t *eh) {
+    return (struct mods_status_state){.mods = zmk_hid_get_explicit_mods()};
+};
+
+void set_mods_symbol(lv_obj_t *label, struct mods_status_state state) {
+    char text[5] = {};
+
+    LOG_DBG("mods changed to %i", state.mods);
+    if (state.mods & (MOD_LCTL | MOD_RCTL))
+        strcat(text, "C");
```

Comment:
Does it make sense to have the display strings for each be Kconfig settings, so folks who are primarily Mac users can use "O" for option, for instance, maybe "CMD" for GUI, etc? Would require some care to determine the string allocation.

Reply:
Thanks for pointing it out, I hadn't considered that indeed. I think constraining each mod to a single character would make our life easier, otherwise we might want a separator between multiple mods as well. It can also help avoid adding too many (4 separate?) Kconfig options. This way we could have a single Kconfig with a string value of 4 chars to correspond to "ctrl-shift-alt-gui" in order. What do you think?

Reply:
That'd work for me, at least as a first pass at this.

Reply:
Sure, I'll take a shot at that and try to add a preprocessor assert if I can to ensure the value has 4 characters.